lib/modules.nix

```
@@ -1488,6 +1488,7 @@ let
     optionName: replacementInstructions:
     { options, ... }:
     {
+      key = "removedOptionModule#" + concatStringsSep "_" optionName;
```

- `#` is commonly used for identifying a component within a resource/document/file, but `removedOptionModule` isn't like a structure of things.

- Let's make the serialization [injective](https://en.wikipedia.org/wiki/Injective_function) with `toJSON`. (This may also be quicker with fewer interpreted function calls, but probably not measurably so)

```suggestion
      key = "removedOptionModule:" + builtins.toJSON optionName;
```

---

Hi @roberth. Thank you for your review and the changes you proposed! Sorry it took some time to get back to it.

Even though I'm not opposed to them, I'd like to understand your position better regarding the both proposed changes, as I have some concerns.

> * `#` is commonly used for identifying a component within a resource/document/file, but `removedOptionModule` isn't like a structure of things.

I was proposing this sign because the option can be considered an entity in a set of options removed with `removedOptionModule`.

> * Let's make the serialization [injective](https://en.wikipedia.org/wiki/Injective_function) with `toJSON`. (This may also be quicker with fewer interpreted function calls, but probably not measurably so)

May I ask what benefits JSON-injectivity will provide, especially when the `key` as a whole isn't JSON-injective?
I see it'll make the key less predictable for humans and will essentially necessitate the use of `toJSON` in `disabledModules`.

```nix
disabledModules = [
     # e.g. this
     { key = "removedOptionModule:" ++ builtins.toJSON [ "boot" "loader" "raspberryPi" ]; }
     # instead of this
     { key = "removedOptionModule#boot_loader_raspberryPi"; }
  ];
```

If going the ultimate the way of JSON-injectivity, we could then even use
```nix
key = builtins.toJSON { removedOptionModule = [ "boot" "loader" "raspberryPi" ]; };
# produces "{\"removedOptionModule\":[\"boot\",\"loader\",\"raspberryPi\"]}"
```
as a `key`. That would also go inline with your suggestion for `:`.
I would really appreciate if you could tell me a couple of words about your motivation, so that we could find even better solution.